lib/fs.js

```diff
@@ -1836,22 +1836,36 @@ fs.FileWriteStream = fs.WriteStream; // support the legacy name
 
 
 WriteStream.prototype.open = function() {
+  const self = this;
+
   fs.open(this.path, this.flags, this.mode, function(er, fd) {
     if (er) {
-      this.destroy();
-      this.emit('error', er);
+      self.destroy();
+      self.emit('error', er);
       return;
     }
 
-    this.fd = fd;
-    this.emit('open', fd);
-  }.bind(this));
```

Comment:
Afaik `bind` is notoriously slow (both at bind time and at call time), so wherever we can remove it (which is everywhere), we most definitely should. Arrow functions, I don't believe perform quite well enough (or prevents optimization?) to start using all over the place I believe. Perhaps @mraleph can chime in on this?